### DISCUSSION TITLE
lazy spawn in keybinding not work

### DISCUSSION THREAD

**Discussion by OriLiMu** [qtile/qtile#4482] [Q&A] [answered]
I have a function like this. 

```python
@staticmethod
def smart_terminal():
    @lazy.function
    def __inner(qtile):
        lazy.cmd_spawn('kate')
        # subprocess.run(["kate"], shell=True)
        # logger.warning(qtile.current_window.name)
        # qtile.current_window.floating = True
    return __inner
```

I want to bind it to a key but not work. kate app not start, but use subprocess works fine how to fix it.

**Comment by tych0** [comment]
I think you want `lazy.spawn` instead of `lazy.cmd_spawn`

**Reply by OriLiMu** [reply]
Same with spawn I try both
```py
    @staticmethod
    def smart_terminal():
        @lazy.function
        def __inner(qtile):
            lazy.cmd_spawn('kate')
            lazy.spawn('kate')
            # process = subprocess.Popen('kate', shell=True)
            # process.wait()
            # logger.warning(qtile.current_window.name)
            # qtile.current_window.floating = True
        return __inner

```

more infomation on log file.

**Reply by tych0** [reply]
Can you post the log file? I don't think you need `staticmethod` and `start_terminal` bits either.

**Reply by tych0** [reply, marked as answer]
Oh, I see, heh. You need to do something like this:
```
def spawn_kate(qtile):
    qtile.cmd_spawn("kate")
Key([mod], "h",              lazy.function(spawn_kate)
```

sorry, hadn't had enough coffee.

**Reply by OriLiMu** [reply]
Hi another thing plz, 
What I want to do is when I am in fullscreen mode. I want to use <kbd>mod p</kbd> to create a float terminal

So here is my step: 
1. start a alacritty first (by spawn just solved by you)
2. then I think to get the current window then set it to float mode will be fine. 
3. But current_window is the prev window not the alacritty you just opened, 
so how to fix it? 

```
    @staticmethod
    def smart_terminal(qtile):
        qtile.cmd_spawn('kate')
        logger.warning(qtile.current_window.name) # show prev window not alacritty
```

**Comment by OriLiMu** [comment]
> Can you post the log file? I don't think you need `staticmethod` and `start_terminal` bits either.

Just this 
```
2023-09-28 22:17:36,162 WARNING libqtile lifecycle.py:_atexit():L33 Restarting Qtile with os.execv(...)

```
